File: seqkit/fa2fq.go
package seqkit

import (
	"ignis/driver/api"
)

type SeqKitFa2FqOptions struct {
	inner Fa2FqOptions
}

type Fa2FqOptions struct {
	Config             KitConfig
	FastaFile          *string
	OnlyPositiveStrand *bool
}

func (this *Fa2FqOptions) setDefaults() *Fa2FqOptions {
	this.Config.setDefaults()
	setDefault(&this.FastaFile, "")
	setDefault(&this.OnlyPositiveStrand, false)

	return this
}

func (this *SeqKitFa2FqOptions) Config(v *SeqKitConfig) *SeqKitFa2FqOptions {
	this.inner.Config = v.inner
	return this
}

func (this *SeqKitFa2FqOptions) FastaFile(v string) *SeqKitFa2FqOptions {
	this.inner.FastaFile = &v
	return this
}

func (this *SeqKitFa2FqOptions) OnlyPositiveStrand(v bool) *SeqKitFa2FqOptions {
	this.inner.OnlyPositiveStrand = &v
	return this
}

func Fa2Fq(input *api.IDataFrame[string], o *SeqKitFa2FqOptions) (*api.IDataFrame[string], error) {
	if o == nil {
		o = &SeqKitFa2FqOptions{}
	}
	opts := o.inner
	opts.setDefaults()
	libprepare, err := api.AddParam(libSource("Fa2Fq"), "opts", OptionsToString(opts))
	if err != nil {
		return nil, err
	}

	return api.MapPartitions[string, string](input, libprepare)
}

File: bigseqkit-py/bigseqkit/fa2fq.py
from bigseqkit.helper import _setDefault, _libSource, _config, _optionsToString, _parseKargs, SeqKitConfig, IDataFrame


class SeqKitFa2FqOptions:

    def __init__(self):
        self.__inner = Fa2FqOptions()

    def config(self, v: SeqKitConfig):
        self.__inner.Config = _config(v)

    def fastaFile(self, v: str):
        self.__inner.FastaFile = v


    def onlyPositiveStrand(self, v: bool):
        self.__inner.OnlyPositiveStrand = v

    def _run(self, input: IDataFrame, **kwargs):
        opts = self.__inner
        _parseKargs(opts, kwargs)
        opts.setDefaults()
        libprepare = _libSource("Fa2Fq").addParam("opts", _optionsToString(opts))
        return input.mapPartitions(libprepare)


class Fa2FqOptions:

    def __init__(self):
        self.Config = None  # KitConfig
        self.FastaFile = None  # str
        self.OnlyPositiveStrand = None  # bool

    def setDefaults(self):
        _setDefault(self, "Config", _config(SeqKitConfig())).setDefaults()
        _setDefault(self, "FastaFile", "")
        _setDefault(self, "OnlyPositiveStrand", False)


def fa2fq(input: IDataFrame, o: SeqKitFa2FqOptions = None, **kwargs):
    if o is None:
        o = SeqKitFa2FqOptions()
    return o._run(input, **kwargs)
